news: announce github issue migration

Swiftb0y opened the pull request:
Still very early state (obviously).
Requesting comments.

Swiftb0y committed:
news: announce github issue migration

incorporate changes requested by review

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

add historical background on mixxx moving away from LP

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

resolve remaining TODOs

fix capitalizations

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

reflow contents to fit 80 char per line limit

add paragraph on githubs involvement

Apply suggestions from code review

Co-authored-by: Jan Holthuis <holthuis.jan@googlemail.com>

reword introduction to use same tense as the body

update blogpost content with recent developments

fixup migration start date

add migration time and closing sentence

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+status: draft
+
+
+Yes everybody, we're officially moving away from Launchpad to Github.
```
```suggestion
Yes everybody, we're moving our bug tracking away from Launchpad to GitHub.
```
The PPA and Ubuntu builds will remain ..

Swiftb0y replied:
Right... but that's not relevant for the majority of contributors, so I would've just said that we're moving away. Also I didn't say we moved completely...

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+
+Yes everybody, we're officially moving away from Launchpad to Github.
+If you've never tried to report a Bug in Mixxx or wanted to contribute
+otherwise, its very likely you have no idea what this Post is about,
```
```suggestion
otherwise, its very likely you have no idea what this post is about,
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+but if you do its even better. No matter if you do or don't, keep reading.
+
+The mixxx team has traditionally been using canonicals (the makers of Ubuntu)
+launchpad platform for hosting their code, managing bug reports and releases.  
```
```suggestion
Launchpad platform for hosting their code, managing bug reports and releases.  
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+
+The mixxx team has traditionally been using canonicals (the makers of Ubuntu)
+launchpad platform for hosting their code, managing bug reports and releases.  
+Over the years, launchpad has become increasingly dated, lack of modern features,
```
```suggestion
Over the years, Launchpad has become increasingly dated, lack of modern features,
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+launchpad platform for hosting their code, managing bug reports and releases.  
+Over the years, launchpad has become increasingly dated, lack of modern features,
+old UI, and bad user experience is the primary reason why many people (myself
+included), especially new potential contributors, stay away from launchpad.
```
```suggestion
included), stay away from Launchpad. Especially for new contributors the mandatory Ubuntu One account was another reason for not to immediately report a bug when it occurred.
```

Swiftb0y replied:
I don't think that's correct. You also need a github account for reporting bugs. The issue is that the bad UI/UX just turns spoiled people off.

daschuer replied:
At least this is the extra burden for new contributors, compared to old. I can remember a user asked me to file a bug for him because if that. But anyway, feel free to adjust you post as you like.

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+Over the years, launchpad has become increasingly dated, lack of modern features,
+old UI, and bad user experience is the primary reason why many people (myself
+included), especially new potential contributors, stay away from launchpad.
+We suspect Launchpad to be one of the major deterents for potential mixxx contributors.
```
Easy to say, hard to prove. Can we delete this sentence, the paragraph above speaks on it's own.

Swiftb0y replied:
Do you mean the entire paragraph should be removed or just the last sentence?

daschuer replied:
Only the last sentence. I think the leak of review resouces is a way bigger issue for contributors. 
The bugs tracker issue is more an issue for experienced users, filing their bugs IMHO. But by removing the sentence, everyone can make their own opinion.. 
That was my idea.

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+old UI, and bad user experience is the primary reason why many people (myself
+included), especially new potential contributors, stay away from launchpad.
+We suspect Launchpad to be one of the major deterents for potential mixxx contributors.
+We've started transitioning away from launchpad quite a while ago.
```
```suggestion
We've started transitioning away from Launchpad quite a while ago.
```

Swiftb0y replied:
Launchpad is an "eigenname" and thus should be capitalized IMO.

daschuer replied:
Right 👍

Swiftb0y replied:
Ah wow, I somehow looked at your suggestions wrong and I thought you lowercased all the capitalized occurrences. Thanks.

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+As I was not around when this happened, most of what I'm writing here is pure speculation
+from my part. Please correct me if I'm wrong or if I should leave this out completely.-->
+
+Back in XXXX we made the first major step away from Launchpad by migrating the
+mixxx source code to github. Still, bug reports continued to be handled on launchpad.
+TODO
```
```suggestion
Back in 2013 we have moved from the Bazaar version control system Git and moved our main branch to GitHub.
Bug reports continued to be handled on Launchpad, because at that time the GitHub's issue implementation where not yet suitable form managing our bug fix and release workflow. This has continuously changed over the years, now GitHub has a duplicate detection feature, can organize bugs in projects and the labeling feature is flexible enough to mimic the Launchpad bug states and keep them after migration.  
```

Swiftb0y replied:
Thank you for the paragraph.

ronso0 replied:
"from the Bazaar version control system **to** Git

Swiftb0y replied:
I rewrote some parts. Please re-review if you don't agree with some changes.

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+TODO
+
+TODO launchpad just used for the ppa, is that correct? no contributor interaction on launchpad
+anymore.
```
correct

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+TODO launchpad just used for the ppa, is that correct? no contributor interaction on launchpad
+anymore.
+
+TODO write about how we migrated all issues to mixxx
```
At one point we make the bug tracking in Launchpad read-only and run the migration. The Lauchpad tags will be migrated 1 to 1. We drop the priorities, because in a volunteer project like Mixxx we have no control about the personal priorities of each contributor.  The remaining Launchpad bug states and priorities will be mapped to special GitHub Lables as in the table below: 

| Launchpad  | GitHub |
| ------------- | ------------- |
| Confirmed | confirmed |
| Fix Committed | (close)  |
| Fix Released | (close) |
| Incomplete | incomplete |
| In Progress | |
| Invalid | invalid |
| New | |
| Triaged | confirmed |
| Won't Fix | wontfix |
| Critical | party stopper; bug |
| High | bug |
| Low | bug |
| Medium | bug |
| Undecided |  |
| Wishlist | feature |

Swiftb0y replied:
I am aware about that table. The question is whether we should mention the "transformations" done as part of the migration in this blogpost.

Swiftb0y replied:
> We drop the priorities, because in a volunteer project like Mixxx we have no control about the personal priorities of each contributor.

I'm not sure if I agree with dropping these, but I'll continue that discussion on zulip.

Swiftb0y replied:
Nevermind, I just saw that we do have labels for the `Critical` and `Wishlist` priorities. Its fine for me that we can drop the rest.

daschuer replied:
> The question is whether we should mention the "transformations" done as part of the migration in this blogpost.

The idea was to use this also as a static reference in case of questions. 
The use case of the new tags after migration are probably best documented in their description field without a reference to the legacy.

Swiftb0y replied:
So do you propose to add the table to this blogpost?

daschuer replied:
Yes

Swiftb0y replied:
Sure

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+From today on, no more bug reports should be filed on launchpad. Currently on-going
+discussion should be continued on Github instead. Note that any comments made
+after 2022-07-07 22:00 UTC have not been migrated to Github.
+<!-- That's when I scraped LP last -->
```
We can make them read only before the final scrape. On the other hand it should be also not much of a problem to add the missing bits.

daschuer reviewed (changes requested):
I have mad some proposals

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+after 2022-07-07 22:00 UTC have not been migrated to Github.
+<!-- That's when I scraped LP last -->
+Existing bug reporters need to create a github account if they haven't already.
+If you have, there is a chance we might already know it and we tagged you in
```
Is this the correct term: 
```suggestion
If you have, there is a chance we might already know it and we will subscribe you to
```

Swiftb0y replied:
Not really. I think technically `@mention` would be correct. Users can configure whether they want to be subscribed to issues/PRs they have been mentioned in. The default is that you are subscribed automatically but that does not need to always be the case.

daschuer replied:
Ok, so let's adjust it like that.

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,41 @@
+<!-- That's when I scraped LP last -->
+Existing bug reporters need to create a github account if they haven't already.
+If you have, there is a chance we might already know it and we tagged you in
+the newly created/migrated issues on Github (sorry for the spam).
```
```suggestion
the newly created/migrated issues on GitHub (sorry for the spam).
```

Swiftb0y commented:
Do we want to publish the blogpost before or after the migration? IMO its somewhat customary to do it before (since the migration is technically a sort of service-outage). Though that comes with its own set of risks because there is a high chance that we won't actually be able to stick to a promised timeline.

Swiftb0y commented:
Done

daschuer commented:
The man issue is probably not the outtake itself, because reporting bugs is nothing we cannot delay or block anyonce work. 
The thing we might want to announce however is the flood of @mentions mails the migration will create. 
Or is there a way to avoid it? Is there a way to issue a warning mail first?

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+
+
+Yes everybody, we're officially moving our bug tracking away from Launchpad to GitHub.
+If you've never tried to report a Bug in Mixxx or wanted to contribute
```
```suggestion
If you've never tried to report a bug in Mixxx or wanted to contribute
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+otherwise, its very likely you have no idea what this post is about,
+but if you do its even better. No matter if you do or don't, keep reading.
+
+The mixxx team has traditionally been using canonicals (the makers of Ubuntu)
```
```suggestion
The mixxx team has traditionally been using Canonical's (the makers of Ubuntu)
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+included), stay away from Launchpad. Especially for new contributors the
+mandatory Ubuntu One account was another reason for not to immediately report
+a bug when they encountered it.
+We've started transitioning away from Launchpad quite a while ago.
```
```suggestion

We've started transitioning away from Launchpad quite a while ago.
```
I am unsure about the blank line below

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+
+Shortly before starting the migration, we made our Launchpad bug tracker
+read-only (some of you might have noticed that). The migration took about XX
+Hours. All of the existing messages were copied to github unfortunately we were
```
```suggestion
Hours. All of the existing messages were copied to GitHub unfortunately we were
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+
+# What does this mean for existing contributors?
+
+From today on, no more bug reports should be filed on launchpad. Currently on-going
```
```suggestion
From today on, no more bug reports should be filed on Launchpad. Currently on-going
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+# What does this mean for existing contributors?
+
+From today on, no more bug reports should be filed on launchpad. Currently on-going
+discussion should be continued on Github instead. Note that any comments made
```
```suggestion
discussion should be continued on GitHub instead. Note that any comments made
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+
+From today on, no more bug reports should be filed on launchpad. Currently on-going
+discussion should be continued on Github instead. Note that any comments made
+after XXXX-XX-XX XX:XX UTC have not been migrated to Github.
```
```suggestion
after XXXX-XX-XX XX:XX UTC have not been migrated to GitHub.
```

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,76 @@
+From today on, no more bug reports should be filed on launchpad. Currently on-going
+discussion should be continued on Github instead. Note that any comments made
+after XXXX-XX-XX XX:XX UTC have not been migrated to Github.
+Existing bug reporters need to create a github account if they haven't already.
```
```suggestion
Existing bug reporters need to create a GitHub account if they haven't already.
```

daschuer reviewed (changes requested):
Thank you. It is almost ready. I have just proposed some capitalization proposals.

Swiftb0y commented:
> Or is there a way to avoid it? Is there a way to issue a warning mail first?

Not sure, I mean we the launchpad and github usernames of all possible users that could mentioned. But actually getting contact info from those is more difficult. I guess I could fixup the script so they don't actually @mention but instead just link their github profile? I already thought about linking the launchpad accounts for all the users we don't know their github handle to.

Swiftb0y commented:
I addressed all your comments and made some further fixes. IMO we just need to fill out the few missing infos (marked with X's) after the migration and then merge ASAP.

daschuer reviewed (approved):
Thank you

daschuer commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,86 @@
+script](https://gist.github.com/Holzhaus/ed384b93465dcc516ae205090e4f179b) which
+made this migration feasible in the first place.
+
+This migration was also only possible because GitHub made a tremendous effort to
+help us. We noticed quite quickly that our import script was going to be
+struggling because of GitHub's secondary rate limiting. After
+[getting in contact with them via
+twitter](https://twitter.com/mixxxdj/status/1546544041038422019) they assembled
+a team of engineers and FOSS experts to help us evaluate possible solutions for
+the rate limitting problem and to avoid email spam from the imported issues.
+Thanks to them, we were able to avoid email spam and reduce the time the import
+took by two orders of magnitude.
```
We may meld this paragraph a bit. Here is a first try: 
```suggestion
Due to the big number of bugs and the flood of related notifications I have noticed quickly that running the script will likely hit various rate limit or spam protection facility.  
After contacting GitHub team via
twitter](https://twitter.com/mixxxdj/status/1546544041038422019), they made a tremendous effort to
help us with a team of engineers and FOSS experts. 
Thanks to them, we were able to avoid email spam and reduce the time the import
took by two orders of magnitude.
```

Swiftb0y replied:
Sure, that works too.

Swiftb0y replied:
To be honest, after trying to work out some of the spelling mistakes and so forth, I'm not very happy how some parts sound. Can you point out what exactly you didn't like about the first version? 

What you do think about this:
```
Due to the big number of bugs and the flood of related notifications, I have
noticed quickly that the script will likely hit various rate limits and spam
protection facilities.  
After [getting in contact with GitHub via
twitter](https://twitter.com/mixxxdj/status/1546544041038422019) they assembled
a team of engineers and FOSS experts to help us evaluate possible solutions.
Thanks to them, we were able to avoid email spam and reduce the time the import
took by two orders of magnitude.
```

daschuer replied:
Perfect. Thank you. 

> Can you point out what exactly you didn't like about the first version?

It was IMHO just too long and detailed compared to the rest of the blog post. That's all.

daschuer replied:
Oh, I just noticed my intend was lost in translation. Ich meinte "zusammenschmelzen"

Swiftb0y replied:
> Perfect. Thank you.

Great. I amended the commit.

Swiftb0y commented:
Does anyone object to me merging this myself after filling in the blanks once the migration is done?

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+title: Migrated to Github issues
```
Captalize all "major" words in the headline (title case).

```suggestion
title: Migrated to GitHub Issues
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+
+Yes everybody, we're officially moving our bug tracking away from Launchpad to
+GitHub. If you've never tried to report a bug in Mixxx or wanted to contribute
+otherwise, it's very likely you have no idea what this post is about, but if you
```
```suggestion
in other ways, it's very likely you have no idea what this post is about, but if you
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+not yet suitable for managing our bugfix and release workflow, thus bug reports
+continued to be handled on Launchpad. This has progressively changed over the
+past couple of years, now GitHub has a "duplicate issue" detection feature, can
+organize bugs in projects and the labeling feature is flexible enough to fillful
```
```suggestion
organize bugs in projects and the labeling feature is flexible enough to fullfil
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+most of our needs.
+
+Going forward, Launchpad will only be used for publishing to our Ubuntu PPA. So
+no contributor outside of the core-team will have to bother with Launchpad
```
```suggestion
no contributor outside of the core team will have to bother with Launchpad
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+organize bugs in projects and the labeling feature is flexible enough to fillful
+most of our needs.
+
+Going forward, Launchpad will only be used for publishing to our Ubuntu PPA. So
```
```suggestion
Going forward, Launchpad will only be used for publishing to our Ubuntu PPA. Hence, 
```

Swiftb0y replied:
the `used` -> `using` change doesn't make sense imo.

Holzhaus replied:
whoops, sorry for the noise.

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+status: draft
+
+
+Yes everybody, we're officially moving our bug tracking away from Launchpad to
```
This is using present progressive, but you use past tense elsewhere. We should use one form consistently.

Swiftb0y replied:
I agree. I'll fix it. past tense makes more sense considering we'll publish this after the migration has been completed.

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+read-only (some of you might have noticed that). The migration took about XX
+Hours. All of the existing messages were copied to GitHub unfortunately we were
+not able to migrate attachments and patches, but since we won't delete the
+launchpad bugtracker, you can still find those attachments there. For future
+reference: decided on the following mapping from Launchpad to Github issue
+labels:
```
```suggestion
launchpad bugtracker, you can still find those attachments there. We
decided on the following mapping from Launchpad to Github issue labels:
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+volunteer is free to work on whatever they choose anyways, we decided to drop
+the different priority levels assigned to issues.  
+
+I want to express a special thanks to our core team member
+[@Holzaus]({author}jan-holthuis) for writing a [custom import
+script](https://gist.github.com/Holzhaus/ed384b93465dcc516ae205090e4f179b) which
+made this migration feasible in the first place.
```
Thanks for the flowers :blush:

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+noticed quickly that the script will likely hit various rate limits and spam
+protection facilities.  
+After [getting in contact with GitHub via
+twitter](https://twitter.com/mixxxdj/status/1546544041038422019) they assembled
```
```suggestion
Twitter](https://twitter.com/mixxxdj/status/1546544041038422019) they assembled
```

Holzhaus commented on content/news/XXXX-XX-XX-github-issues-migration.md:
```diff
@@ -0,0 +1,85 @@
+# What does this mean for existing contributors?
+
+From today on, no more bug reports should be filed on Launchpad. Currently
+on-going discussions should be continued on GitHub instead. Note that any
```
```suggestion
ongoing discussions should be continued on GitHub instead. Note that any
```

Holzhaus reviewed (commented):
Thanks for writing this. Some minor comments.

Swiftb0y commented:
made some more elaborate changes since some of the info actually outdated by now. https://github.com/mixxxdj/website/pull/274/commits/62b92aa4f307fdbc6032234e9382989645d83393

Swiftb0y commented:
squashed. I'll merge in an hour or so.
